cst438_register/src/main/java/com/cst438/service/GradebookServiceMQ.java

```diff
@@ -43,6 +47,18 @@ public void receive(CourseDTOG courseDTOG) {
 		
 		//TODO  complete this method in homework 4
 		
+		CourseDTOG.GradeDTO gradeDTO = new CourseDTOG.GradeDTO();
```

Comment:
Same feedback as with the REST service. I think you need a loop here and you won't need gradeDTO. This object is empty, so there's no info in it like student_email, etc. Everything you need should be in courseDTOG.